src/widget/woverview.cpp

```diff
@@ -169,12 +169,27 @@ void WOverview::slotWaveformSummaryUpdated() {
         return;
     }
     m_pWaveform = pTrack->getWaveformSummary();
-    // If the waveform is already complete, just draw it.
-    if (m_pWaveform && m_pWaveform->getCompletion() == m_pWaveform->getDataSize()) {
-        m_actualCompletion = 0;
-        if (drawNextPixmapPart()) {
-            update();
+    if (m_pWaveform) {
+        // If the waveform is already complete, just draw it.
+        if (m_pWaveform->getCompletion() == m_pWaveform->getDataSize()) {
+            m_actualCompletion = 0;
+            if (drawNextPixmapPart()) {
+                update();
+            }
+        }
+    } else {
+        // Null waveform pointer means waveform was cleared.
+        if (m_pWaveformSourceImage) {
+            delete m_pWaveformSourceImage;
```

Comment:
I have decided to take a (slightly) different approach with this and make this QImage statically allocated (like `m_waveformImageScaled` is). I hope you don't mind. :smiley: